### PR TITLE
🍰 Feature/auth

### PR DESCRIPTION
**Pull request by memhard**
## What changed?
add jwt autharization, change models, add user registration and etc changes...     

## Issue ticket(s) number and link     
- Resolves No link 
     
## Checklist before requesting a review     
- [x] I have performed a self-review of my code           
- [x] Code is functional on my machine(Windows 11/Ubuntu22.04 LTS)

### CODE REVIEW

File: src/auth/manager.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    # generate new access token
+    access_tkn = create_access_jwt(data)
+    return {
+        'access_token': access_tkn,
+        'refresh_token': refresh_tkn,
+        'type': 'bearer'
+    }
+
+
+async def verified_user(token: str = Depends(oauth_scheme), session: AsyncSession = Depends(get_async_session)) -> User:
```
Comment (Olderestin) [inline src/auth/manager.py]: Change func name

File: .env.example [hunk 2]
```diff
@@ -0,0 +1,9 @@
+POSTGRES_USER = postgres
+POSTGRES_PASSWORD = postgres
+POSTGRES_DB = postgres
+POSTGRES_PORT = 5432
+POSTGRES_HOST = localhost
+
+SECRET = secret_key
+ALGORITHM = HS256
+
```
Comment (Olderestin) [inline .env.example:9]:
```suggestion
PGADMIN_DEFAULT_EMAIL = 'postgres@postgres.com'
PGADMIN_DEFAULT_PASSWORD = 'postgres'
```
Add pgadmin settings

**Review by Olderestin — COMMENTED** [review]
Make changes and can aprove 👍

File: src/auth/routers.py [hunk 3]
```diff
@@ -1,3 +1,96 @@
...
+    Args:
+        token_data: The token data
+    
+    Returns:
+        The token data
+    """
+    return token_data
+
+@auth_router.post('/logout')
+async def logout(data: str = Depends(get_token_data), session: AsyncSession = Depends(get_async_session)):
```
Comment (Olderestin) [inline src/auth/routers.py]: I think we can use current_user Depends already

File: requirements.txt [hunk 4]
```diff
@@ -1,6 +1,51 @@
-fastapi[all]
-asyncpg
-sqlalchemy
-alembic
-beautifulsoup4
-requests
```
Comment (Olderestin) [inline requirements.txt]: Why do you change requirements?

File: src/auth/manager.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        session: The database session
+        
+    Returns:
+        A dictionary with the new access,refresh tokens and the token type
+    """
+    
+    if data['mode'] != 'refresh_token':
+        raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail='not a refresh token'
```
Comment (Olderestin) [inline src/auth/manager.py:92]:
```suggestion
                detail='Not a refresh token'
```
Does it metter for front if they will wait for messages starting with capital letter?

File: src/auth/manager.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        session: The database session
+    
+    Returns:
+        The user associated with the token
+    """
+    
+    if data['mode'] != 'access_token':
+        raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail='not an access token'
```
Comment (Olderestin) [inline src/auth/manager.py:132]: Same

File: src/auth/manager.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+            )
+    
+    stmt = select(User).where(User.email == data['user_email'])
+    result = await session.execute(stmt)
+    user = result.scalars().first()
+        
+    if not user:
+        raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail='invalid authorization credentials'
```
Comment (Olderestin) [inline src/auth/manager.py:142]: Same

File: src/auth/manager.py [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+            )
+    
+    if not user.refresh_token:
+        raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail='token has been revoked'
```
Comment (Olderestin) [inline src/auth/manager.py:148]: Maybe we don't need to send details about it, just <<invalid credentials>>?

**Review by Olderestin — CHANGES_REQUESTED** [review]
Need changes.

File: src/auth/routers.py (hunk 3 above)
Comment (memhard) [reply src/auth/routers.py]: fixed

**Review by memhard — COMMENTED** [review]
tasks done

**Pull request merged**